### PR TITLE
Prevent TestResult#toString() from throwing exceptions.

### LINKED ISSUE #1298: ITestResult injection is failing in BeforeMethod method
From https://github.com/cbeust/testng-eclipse/issues/307

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes #1298

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @cbeust @juherr @missedone

**Review by juherr — COMMENTED** [review]
Could we add an enum for status and deprecate old values?

**Comment by krmahadevan** [comment]
@juherr - Currently test statuses are represented as int values and they seem to be used in `ITestResult#getStatus` and `ITestResult#setStatus`. Changing the status to an enum, would also require that we change the status setting/getting mechanisms across TestNG (from working with an `int` to working with a new `enum`). Is that really required ? Without doing that change, I don't see a lot of value add in just changing the status type from int to enum.

File: src/main/java/org/testng/internal/TestResult.java [hunk 1]
```diff
@@ -230,7 +231,8 @@ private static String toString(int status) {
       case SKIP: return "SKIP";
       case SUCCESS_PERCENTAGE_FAILURE: return "SUCCESS WITHIN PERCENTAGE";
       case STARTED: return "STARTED";
-      default: throw new RuntimeException();
+      case CREATED: return "CREATED";
+      default: throw new TestNGException("Encountered an un-defined test status of [" + status + "].");
```
Comment (missedone) [inline src/main/java/org/testng/internal/TestResult.java]: i'd let default to return something `un-defined test status of [" + status + "]`, the point is `toString()` should be as quiet as possible.
however, now that the case branches cover all the status values, so i'm fine with this.

**Comment by missedone** [comment]
👍

**Comment by juherr** [comment]
@krmahadevan the only value add I see is avoiding similar issues in the future ;)
But I'm agree, it is not really important.

File: src/main/java/org/testng/internal/TestResult.java (hunk 1 above)
Comment (juherr) [inline src/main/java/org/testng/internal/TestResult.java]: With an enum, it could store the string value itself. ;p

**Comment by krmahadevan** [comment]
@juherr - I agree with the benefits of enum over int (no second thoughts on that). Please let me know the final verdict on this :) Is this PR good to be merged or is there still something that needs to be done before it can be merged ?

**Comment by juherr** [comment]
@krmahadevan I prefer enum but it is not a blocking point.

**Review by juherr — APPROVED** [review]
LGTM with or without enum 👍 Ping @cbeust

File: src/main/java/org/testng/internal/TestResult.java (hunk 1 above)
Comment (cbeust) [inline src/main/java/org/testng/internal/TestResult.java]: I was thinking about converting these to `enum` as well but they are defined on `ITestResult`, which is part of the API, so I'm not 100% sure it's a backward compatible change.

**Pull request merged**